### ISSUE TITLE
Help understanding how to use this toolbox

### ISSUE THREAD

**Issue by dbuscombe-usgs** [TsLu1s/mlimputer#1]
Hi, thanks for this toolbox - it looks really useful. But I'm struggling to understand a few things about how you've set up the problem.

In your examples, you are fitting models and cross-validating on a train subset, but the test subset never seems to get used. How do you know how well the model performs on the test subset? Otherwise, what is the point of the test subset? You make it, but don't use it. So, it seems you dont know how well your models truly perform.

What I was expecting to see was the following steps:

1. split a dataset into train and test
2. copy the test subset
3. introduce nans to the copy of the test subset (only - what would be the point of introducing nans to the train subset?)
4. make a model using the train subset
5. apply that model to the test subset
6. compare that imputed test subset with the original test values (before nans were introduced)

Anything else I'm really struggling to understand. Isn't this the only way to know how well the model works? Why is the cross-validation function not aware of the test subset? 

Also, it seems that you set the problem up by putting nans in the predictor variables, but not the target variable. Shouldn't the model use the predictor variables to impute nans in the target variable only? This is most confusing.

Any light you can shed on this would be helpful. A fully worked example for how you might read in a time-series to impute missing variables in the target variable, then testing the model to see how well it works, would be helpful. Otherwise, I'm just struggling to see what this toolbox is really for.

Thanks

**Comment by TsLu1s** [comment]
Hi mate,

I lament for my delay in replying, only saw it now.
So, i understand some of your doubts\confusion in regards to the logical structure of the evaluation done, primarily since there was a parameter in the cross validation function 'test_size' which was kinda unproperly implemented since it was really a split size/validation size by cross validation folder.

The test was not being use, mainly because the cross validation was there to get the sense of the basic performance of the null imputation models, if you wanted you could absolutely have used the test set and evaluate the test performance after each the cross validation analysis.

Lastly, i think there is a bit of a mismatch in terms of what you think this framework does and what it really does. This framework main purpose in to impute numerical missing values in the input columns (all other then the to be predicted target) through various advanced supervised ML algorithmic techniques\options. This is not to predict null values in the target variable, as that would be classical classification or regression contexts, think of this framework as a preprocessing\automated null treatment framework.

Also: I've made a considerable refactor in the project just now, that i think it solved all your referenced doubts. Feel free to check the examples folder and comeback to me if my answer & new solutions aren't still quite clear to you.

Thanks a lot for your feedback :)

**Comment by dbuscombe-usgs** [comment]
I'm not sure what problem this solves, the examples promote bad practices, and your garbled response is unhelpful.

**Comment by dbuscombe-usgs** [comment]
Now you may close this issue. Now I've responded

**Comment by TsLu1s** [comment]
Missing data imputation is a very relevant procedure in data analysis in generaly, especially in ML & Supervised ML, as most models do not deal nans in the training data. In simple terms this is what this framework tackles in an more advanced and refined way, that's it.

Thank you for your feedback once more :)
